archaius-etcd/src/main/java/com/netflix/config/source/EtcdConfigurationSource.java

```diff
@@ -129,6 +129,11 @@ public void handle(Response updateResponse) {
                         updateConfiguration(createIncremental(null, ImmutableMap.<String, Object>of(sourceKey, value), null));
                         break;
 
+                    case "update":
```

Comment:
Do you want to check first if sourceKey exists when doing "Update"? If not, you can either throws some exception or change to Set? I'm not familiar with etcd module. I'll leave the PR open for now and hopefully @spoon16 can take a look. Also, I probably won't be able to release a new Archaius until mid Jan.